Electron/Core/RoveProtocol/Rovecomm.ts

```diff
@@ -126,35 +134,120 @@ function parse(packet: Buffer): void {
   }
 }
 
-function TCPListen(socket: Socket) {
+function TCPParseWrapper(socket: TCPSocket) {
+  /*
+   * Takes in an object of the TCPSocket type defined in the interface TCPSocket at the top of this file
+   * Iterates while there is still at least five bytes in the Deque in the TCPSocket, parsing one RC Packet at a time
+   * Returns when there is either less than 5 bytes in the Deque or not a complete packet in the Deque
+   */
+  // While the Deque contains at least a header to allow parsing control packets
+  while (socket.RCDeque.length >= 5) {
+    const dataCount = socket.RCDeque.get(3)
+    const dataSize = dataSizes[socket.RCDeque.get(4)]
+    // If the length of the Deque is more than the header size and the size of the packet, make a buffer and then parse that buffer
+    if (socket.RCDeque.length >= 5 + dataCount * dataSize) {
+      // create another list to put the entire packet into
+      const packet = []
+      for (let i = 0; i < 5 + dataCount * dataSize; i++) {
+        // Here we use Shift to get and remove the elements that make up the packet to prevent parsing the same packet multiple times
+        packet.push(socket.RCDeque.shift())
+      }
+      // Make a buffer from that list, needed to provide the correct input for the parse function
+      const packetBuffer = Buffer.from(packet)
+      parse(packetBuffer)
+      // If the Deque doesn't contain a full packet, return
+    } else return
+  }
+}
+
+function TCPListen(socket: TCPSocket) {
   /*
-   * Listens on the passed in TCP socket, always calling parse if it recieves anything
+   * Listens on the passed in TCP socket, pushing received data into the TCPSocket Deque and then calling the TCP Parse Wrapper when data is received
    */
-  socket.on("message", (msg: Buffer) => {
-    parse(msg)
+  const selfComp = 1
+  socket.RCSocket.on("data", (msg: Buffer) => {
+    // eslint-disable-next-line no-restricted-syntax
+    for (let i = 0; i < msg.length; i++) {
+      // use this if compared to self to get rid of the linter no self compare
+      // eslint-disable-next-line no-self-compare
+      if (selfComp === selfComp) {
```

Comment:
this const selfComp is essentially just an if true, which seems really dumb. This if seems useless.